### PR TITLE
bump base64 from 0.10 to 0.21

### LINKED ISSUE #26013: Deduplicate base64
Latest release is 0.12, brought in crate tree through #26005.

Dependencies that already depend on 0.12 that we need to bump on our side: 
 * `headers`
   * `warp`
   * `hyper_serde`

Dependencies on 0.10 that Mozilla people maintain:
 * `webdriver` (depends on `warp`)
 * `webrender`

Dependencies on 0.10 that require upstream changes:
 * `influent`

Dependencies on 0.11 that require upstream changes:
* [`rust-argon2`](https://github.com/sru-systems/rust-argon2/issues/32)
  * [`redox_users`](https://gitlab.redox-os.org/redox-os/users/issues/30)

### PR DESCRIPTION
**Pull request by dvtkrlbs**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes partly fix #26013 
- [x] These changes do not require tests because it bumps version of a dependency. From running the tests there does not seem to be any regressions (there are a lot of crashed tests but I don't think they are relevant to this change)

### CODE REVIEW

**Comment by dvtkrlbs** [comment]
I looked at the dependency tree using cargo tree with -i flag and one of the dependencies is content-security-policy and other is headers and they seem reluctant to upgrade base64 based on MSRV concerns. THere is also ron (one for 0.10.1 and one for 0.13 with different ron versions) and webdriver which I did not follow the conversation for base64.

**Comment by jdm** [comment]
@bors-servo r+
Thanks! There's also a new version of content-security-policy released (0.5.1) with a newer base64 which we can update to.

**Comment by dvtkrlbs** [comment]
I used `cargo update -p content-security-policy` to update the lock version. The diff for cargo.lock does not seem to show any changes related to content-security-policy but it seems it got updated.

File: Cargo.lock [hunk 1]
```diff
@@ -5043,8 +5049,13 @@ dependencies = [
...
  "atomic_refcell",
  "backtrace",
+<<<<<<< Updated upstream
  "base64 0.21.2",
  "bitflags",
+=======
+ "base64 0.10.1",
+ "bitflags 1.3.2",
+>>>>>>> Stashed changes
  "bluetooth_traits",
```
Comment (jdm) [inline Cargo.lock]: This merge conflict needs to be resolved. Also https://github.com/servo/servo/blob/master/servo-tidy.toml#L25 will need a new entry for `bitflags`.

Comment (dvtkrlbs) [reply Cargo.lock]: Oh sorry I missed when switching the branch. I force pushed fixing this.

Comment (dvtkrlbs) [reply Cargo.lock]: wait i missed an another line

Comment (dvtkrlbs) [reply Cargo.lock]: fixed

**Comment by jdm** [comment]
Hmm, we've broken some tests with this change:
```
OK /html/webappapis/atob/base64.any.html

    FAIL [expected PASS] subtest: btoa("ab") == "YWI=" The string contains invalid characters.
    FAIL [expected PASS] subtest: btoa("abcd") == "YWJjZA==" The string contains invalid characters.
    FAIL [expected PASS] subtest: btoa("abcde") == "YWJjZGU=" The string contains invalid characters.
    FAIL [expected PASS] subtest: btoa("\0a") == "AGE=" The string contains invalid characters.
    FAIL [expected PASS] subtest: btoa(null) == "bnVsbA==" The string contains invalid characters.
    FAIL [expected PASS] subtest: btoa(7) == "Nw==" The string contains invalid characters.
    FAIL [expected PASS] subtest: btoa(12) == "MTI=" The string contains invalid characters.
    FAIL [expected PASS] subtest: btoa(true) == "dHJ1ZQ==" The string contains invalid characters.
    FAIL [expected PASS] subtest: btoa(false) == "ZmFsc2U=" The string contains invalid characters.
    FAIL [expected PASS] subtest: btoa(Infinity) == "SW5maW5pdHk=" The string contains invalid characters.
    And 273 more unexpected results...
```
To reproduce, run `./mach test-wpt tests/wpt/web-platform-tests/html/webappapis/atob/base64.any.html`

**Comment by dvtkrlbs** [comment]
This seems really weird. I did a small test and the encoded string is same byte by byte between the two versions.

**Comment by dvtkrlbs** [comment]
It seems that the actual assert succeeds but there is an additional warning that causes the test to fail. (Clicking on the test case shows the assert itself)

**Comment by dvtkrlbs** [comment]
I did a few more test using the firefox devtools and the code tested does not seem to cause any problems. Is there an easy way to attach a devtools session to a test-wpt run i could not find a simple flag to be able to attach to the process from firefox ? Other than that I am at a lock here I am not sure about the cause of the regression.

**Comment by mrobinson** [comment]
Trying to unblock this: I pushed a new commit to this branch which adjusts the configuration of the base64 engine to not require padding on decode. I think this fixes the tests.

@bors-servo try=wpt

**Review by mrobinson — APPROVED** [review]

**Comment by mrobinson** [comment]
CI failed because of #30063.

**Comment by mrobinson** [comment]
CI failed due to #30064.

**Pull request merged**